### PR TITLE
Issue1630: Fixing failing slow tests due to object creation expressions

### LINKED ISSUE #1630: Fix slow tests
PR #1524 broke the slow tests. 

```java
[ERROR] Tests run: 20, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 13.86 s <<< FAILURE! - in com.github.javaparser.symbolsolver.resolution.AnalyseJavaSymbolSolver060Test
[ERROR] parseCoreJavaparsermodelDeclarations(com.github.javaparser.symbolsolver.resolution.AnalyseJavaSymbolSolver060Test)  Time elapsed: 1.902 s  <<< FAILURE!
org.junit.ComparisonFailure: Line 21 of /home/danny/IdeaProjects/javaparser/javaparser-symbol-solver-testing/src/test/test_sourcecode/javasymbolsolver_0_6_0/expected_output/java-symbol-solver-core/com_github_javaparser_symbolsolver_javaparsermodel_declarations_JavaParserInterfaceDeclaration.txt is different from what is expected expected:<...).asInterface() ==> [com.github.javaparser.symbolsolver.model.declarations.TypeDeclaration.asInterface()]> but was:<...).asInterface() ==> [ERROR]>
	at com.github.javaparser.symbolsolver.resolution.AnalyseJavaSymbolSolver060Test.parse(AnalyseJavaSymbolSolver060Test.java:127)
	at com.github.javaparser.symbolsolver.resolution.AnalyseJavaSymbolSolver060Test.parseCoreJavaparsermodelDeclarations(AnalyseJavaSymbolSolver060Test.java:200)

[ERROR] parseModelTypesystem(com.github.javaparser.symbolsolver.resolution.AnalyseJavaSymbolSolver060Test)  Time elapsed: 0.717 s  <<< FAILURE!
org.junit.ComparisonFailure: Line 119 of /home/danny/IdeaProjects/javaparser/javaparser-symbol-solver-testing/src/test/test_sourcecode/javasymbolsolver_0_6_0/expected_output/java-symbol-solver-model/com_github_javaparser_symbolsolver_model_typesystem_ReferenceType.txt is different from what is expected expected:<...Values().get(i) ==> [java.util.List.get(int)]> but was:<...Values().get(i) ==> [ERROR]>
	at com.github.javaparser.symbolsolver.resolution.AnalyseJavaSymbolSolver060Test.parse(AnalyseJavaSymbolSolver060Test.java:127)
	at com.github.javaparser.symbolsolver.resolution.AnalyseJavaSymbolSolver060Test.parseModelTypesystem(AnalyseJavaSymbolSolver060Test.java:331)
```

### PR DESCRIPTION
**Pull request by ftomassetti**
Fix #1630 

This test is failing locally as it was before my changes:

```
SymbolSolverCollectionStrategyTest.resolveExpressions:28->lambda$resolveExpressions$0:31 » NoClassDefFound
```

Any advice?

### CODE REVIEW

**Comment by ftomassetti** [comment]
I should add a few javadocs and a few tests for supporting methods I introduced

**Comment by ftomassetti** [comment]
Starting by adding javadocs

File: javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/ObjectCreationContext.java [hunk 1]
```diff
@@ -73,6 +74,9 @@ public ObjectCreationContext(ObjectCreationExpr wrappedNode, TypeSolver typeSolv
 
     @Override
     public SymbolReference<ResolvedMethodDeclaration> solveMethod(String name, List<ResolvedType> argumentsTypes, boolean staticOnly, TypeSolver typeSolver) {
+        //ResolvedType resolvedType = JavaParserFacade.get(typeSolver).getType(wrappedNode);
+        //List<ResolvedMethodDeclaration> possibleMethods = resolvedType.asReferenceType().getAllMethods();
+        //return MethodResolutionLogic.findMostApplicable(possibleMethods, name, argumentsTypes, typeSolver);
```
Comment (matozoid) [inline javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/ObjectCreationContext.java]: Work in progress? :-)

Comment (ftomassetti) [reply javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/ObjectCreationContext.java]: Committed by mistake together with Javadocs :( removing

**Comment by ftomassetti** [comment]
Testing getDirectAncestors, remain to test getAllMethods

**Comment by matozoid** [comment]
Hey @ftomassetti - is this complete now?

**Comment by ftomassetti** [comment]
Yes, it was complete but adding tests for getAllMethods would be a good thing to do

**Pull request merged**